components/Buttons/src/MDCFloatingButton.h

```diff
@@ -96,6 +118,32 @@ typedef NS_ENUM(NSInteger, MDCFloatingButtonShape) {
 
 - (nullable instancetype)initWithCoder:(nonnull NSCoder *)aDecoder NS_DESIGNATED_INITIALIZER;
 
+- (void)setMinimumSize:(CGSize)minimumSize NS_UNAVAILABLE;
+
+- (void)setMinimumSize:(CGSize)minimumSize forShape:(MDCFloatingButtonShape)shape
+    UI_APPEARANCE_SELECTOR;
+
+- (void)setMaximumSize:(CGSize)maximumSize NS_UNAVAILABLE;
+
+- (void)setMaximumSize:(CGSize)maximumSize forShape:(MDCFloatingButtonShape)shape
+    UI_APPEARANCE_SELECTOR;
+
+- (void)setContentEdgeInsets:(UIEdgeInsets)contentEdgeInsets NS_UNAVAILABLE;
+
+- (void)setContentEdgeInsets:(UIEdgeInsets)contentEdgeInsets forShape:(MDCFloatingButtonShape)shape
+    UI_APPEARANCE_SELECTOR;
+
+- (void)setHitAreaInsets:(UIEdgeInsets)hitAreaInsets NS_UNAVAILABLE;
+
+- (void)setHitAreaInsets:(UIEdgeInsets)insets forShape:(MDCFloatingButtonShape)shape
+    UI_APPEARANCE_SELECTOR;
+
+- (void)setElevation:(CGFloat)elevation forState:(UIControlState)state NS_UNAVAILABLE;
```

Comment:
This means we are going to have to overload the MDCButton sets the elevation when state changes to take into account our shape.